### DISCUSSION TITLE
How is the new user registration process expressed

### DISCUSSION THREAD

**Discussion by mwpowellhtx** [OrchardCMS/OrchardCore#11168] [Q&A]
Hello, apologies if this question has been asked in one form or another. We are also curious about the new user registration process, perhaps it is a workflow, I don't know yet exactly. Customer visits our business web site, says she wants to shop at our web site, creates a new registration. Let's say we are integrated with an Amazon Cognito style provider, all completely external from our site. OAuth callbacks occur and eventually lands back on our web site with a 'registered' client ID and access token. What happens next from an Orchard and/or CMS perspective? Without necessarily needing to manually intervene, [manage](https://docs.orchardcore.net/projects/O1/en/latest/Documentation/Managing-users-and-roles), etc.

**Comment by Skrypt** [comment]
/cc @kevinchalet

I believe this is something that can be achieved using **OpenId Connect** with the OpenIddict module in Orchard Core. Though, I don't know details about the Amazon Cognito service.

**Comment by Skrypt** [comment]
I moved this to the discussion board of Orchard Core. I believe the question though was about Orchard 1. I cannot give a clear answer for that as I don't think we had an OpenId module back then. So, this can be achieved now with Orchard Core. Also, assumed the question was for Orchard 1 because of the documentation link. Else, if it was for Orchard Core then you are looking at the wrong documentation.

**Comment by Skrypt** [comment]
Also, you can use https://github.com/aws/aws-aspnet-cognito-identity-provider for ASP.NET Core.

**Comment by mwpowellhtx** [comment]
> May I ask why you asked this question in the OrchardDoc repository? This repository is for the Orchard 1 documentation.

My mistake. I did not notice that until just now.
